components/sensor/sds011.rst

```diff
@@ -0,0 +1,81 @@
+SDS011 Particulate Matter Sensor
+================================
+
+.. seo::
+    :description: Instructions for setting up SDS011 Particulate matter sensors
+    :image: sds011.jpg
+    :keywords: sds011
+
+The ``sds011`` sensor platform allows you to use SDS011 particulate matter sensors (`datasheet <https://nettigo.pl/attachments/398>`__)
+with ESPHome.
+
+.. figure:: images/sds011-full.jpg
+    :align: center
+    :width: 50.0%
+
+    SDS011 Particulate Matter Sensor
+
+As the communication with the SDS011 is done using UART, you need
+to have an :ref:`UART bus <uart>` in your configuration with the ``rx_pin`` connected to the TX pin of the
+SDS011 and optionally the ``tx_pin`` connected to the RX Pin of the SDS011 (it's switched because the
+TX/RX labels are from the perspective of the individual device). Additionally, you need to set the baud rate to 9600.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      rx_pin: D0
+      tx_pin: D1
+      baud_rate: 9600
+
+    sensor:
+      - platform: sds011
+        pm_2_5:
+          name: "Particulate Matter <2.5µm Concentration"
+        pm_10_0:
+          name: "Particulate Matter <10.0µm Concentration"
+        update_interval: 10min
+        rx_only: false
+
+With ``update_interval``, the working period of the SDS011 device will be changed. If ``update_interval`` is less
+than 1 minute, the SDS011 will be set to continuous measurement and will report new measurement values
+approximately every second.
+
+If ``update_interval`` is between 1-30 minutes, the SDS011 periodically turns on for 30s before each measurement.
+For the remaining time the sensor is shut off. As a result, this mode can reduce power consumption and increases
+the lifetime of the SDS011.
+
+Configuration variables:
+------------------------
+
+- **pm_2_5** (*Optional*): Use the concentration of particulates of size less than 2.5µm in µg per cubic meter.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **pm_10_0** (*Optional*): Use the concentration of particulates of size less than 10.0µm in µg per cubic meter.
+  All options from :ref:`Sensor <config-sensor>`.
+
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. This affects the working
+  period of the SDS011 sensor. Defaults to ``1s``.
```

Comment:
1s is not really right. The default value is 0 (in terms of configuration value), but the behavior is 1s.

Reply:
I see. What about changing the default update_interval to something like 5min?

The thought behind that is: If times >1min are preferred because of sensor lifetime, let's make that the default and users can still change it later.